finat/point_set.py

```diff
@@ -0,0 +1,123 @@
+from __future__ import absolute_import, print_function, division
+from six import with_metaclass
+from six.moves import range
+
+from abc import ABCMeta, abstractproperty
+from itertools import chain, product
+
+import numpy
+
+import gem
+from gem.utils import cached_property
+
+
+class AbstractPointSet(with_metaclass(ABCMeta)):
+    """A way of specifying a known set of points, perhaps with some
+    (tensor) structure."""
+
+    @abstractproperty
+    def points(self):
+        """A flattened numpy array of points with shape
+        (# of points, point dimension)."""
```

Comment:
We use this to flatten the points for backwards compatibility with FIAT elements.